### PR TITLE
Adds dropdowns to breadcrumbs

### LINKED ISSUE #1212: provide a breadcrumb navigation to other pages in a plugin
**Describe the problem/challenge you have**
Inside the [octant-jx](https://github.com/jenkins-x/octant-jx) we have a number of links to other pages (some inside the octant-jx plugin, some inside octant itself; e.g. the pod view for a tekton pipeline) - it would be nice to have quick way to navigate to them from the breadcrumb bar.

Similarly it'd be nice to be able to link to other pages; maybe a right aligned drop down menu for related views. e.g. when viewing a Pipeline in octant-jx it'd be nice to navigate to the Logs, the Pod, the Step definition, the Pull Request that triggered the pipeline, the preview application etc. There's not an obvious easy canonical way to have menus/links on a page to other related areas.

**Describe the solution you'd like**
Allowing a breadcrumb bar to have a little pull down menu so you can quickly navigate from one resource to another. E.g. on Workloads it might be nice to navigate to Deployments / Jobs quickly from a down arrow next to Workloads (without having to find where you are in the big left hand nav bar)?

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Octant version (use `octant version`): 
- Kubernetes version (use `kubectl version`):
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc):

### PR DESCRIPTION
**Pull request by mklanjsek**
Where applicable, breadcrumbs will now contain dropdown controls, providing additional way to navigate directly from the breadcrumb bar.

Most notable changes:

- Consolidated breadcrumb functionality inside a single go class,
- Added logic to place dropdowns inside breadcrumbs for all object and custom resource types,
- Added logic to create breadcrumbs only for top level components, we need it now for performance reasons,
- Fixed inconsistencies in structuring breadcrumbs, mostly in cluster module,
- Fixed naming issues that were preventing dropdown selection,
- Added `Selection` property to dropdown that can be used to specify selected item, 

Signed-off-by: Milan Klanjsek <mklanjsek@pivotal.io>

**Which issue(s) this PR fixes**
- Fixes #1212

### CODE REVIEW

**Comment by GuessWhoSamFoo** [comment]
Just jumped off a call with @mklanjsek , most of the review comments here are likely going to be irrelevant with suggested refactors.

File: internal/describer/breadcrumb.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	b := &Breadcrumb{
+		BreadcrumbConfig: bc,
+	}
+	return b
+}
+
+func (bc BreadcrumbConfig) FromObject(root ResourceLink, parent ResourceLink, items *unstructured.UnstructuredList) []component.TitleComponent {
+	var title []component.TitleComponent
+
+	if strings.Contains(root.Url, "($NAMESPACE)") {
```
[the remark this answers is not in the document]
Comment (mklanjsek) [reply internal/describer/breadcrumb.go:44]: Addressed in new approach where breadcrumb info is gathered from module navigation data.

File: internal/describer/object_test.go [hunk 2]
```diff
@@ -110,7 +110,7 @@ func TestObjectDescriber(t *testing.T) {
 			)))
 
 	expected := component.ContentResponse{
-		Title: component.Title(component.NewLink("", "object", "."), component.NewText("pod")),
+		Title: component.Title(component.NewText("pod")),
```
Comment (GuessWhoSamFoo) [inline internal/describer/object_test.go:113]: https://github.com/vmware-tanzu/octant/pull/1716/files#diff-f9af2246ccbdfd0724c02744f100700f43dd56d59254ac905a9a42d92b8ddba5L60 isn't meaningful at anytimes here

**Review by GuessWhoSamFoo — CHANGES_REQUESTED** [review]
Some minor changes - good work!

File: internal/api/breadcrumb.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	return title
+}
+
+// Create first part of bredcrumb from module navigation entries.
+// Performs reverse path traversal and creates all related breadcrumbs.
+func CreateNavigationBreadcrumb(navs []navigation.Navigation, contentPath string) (LinkDefinition, []component.TitleComponent) {
+	var last LinkDefinition
+	var title []component.TitleComponent
+
+	thisPath := contentPath
```
[the remark this answers is not in the document]
Comment (mklanjsek) [reply internal/api/breadcrumb.go:78]: Yes!  It's used as title for next element in breadcrumb...

File: internal/api/breadcrumb.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+			second := dropdownFromList(parent, path.Base(contentPath), list)
+			title = append(title, second, component.NewText(path.Base(contentPath)))
+		} else {
+			title = append(title, component.NewText(parent.Title))
+		}
+	}
+	return title
+}
+
+// Create first part of bredcrumb from module navigation entries.
```
[the remark this answers is not in the document]
Comment (mklanjsek) [reply internal/api/breadcrumb.go]: THanks 😊

File: internal/api/breadcrumb.go [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	}
+
+	if strings.Contains(contentPath, crPath) {
+		if path.Base(contentPath) == parent.Title || path.Base(parent.Url) == crPath {
+			title = append(title, component.NewText(parent.Title))
+		} else {
+			title = append(title, component.NewLink("", parent.Title, parent.Url), component.NewText(path.Base(contentPath)))
+		}
+	} else {
+		gavk, err := cm.moduleManager.GvkFromPath(path.Dir(contentPath), state.GetNamespace())
```
[the remark this answers is not in the document]
Comment (mklanjsek) [reply internal/api/breadcrumb.go]: Renamed.

File: internal/describer/object_test.go (hunk 2 above)
Comment (mklanjsek) [reply internal/describer/object_test.go:113]: I wanted to keep that test because that's still used to set title in some cases for example table headers.

**Comment by mklanjsek** [comment]
@GuessWhoSamFoo PTAL, I made changes or commented on your points and squashed all commits. 

Thanks for reviewing it so quickly!

**Review by GuessWhoSamFoo — APPROVED** [review]

**Pull request merged**